src/robot/parsing/model/statements.py

```diff
@@ -417,6 +759,23 @@ def args(self):
 class ForHeader(Statement):
     type = Token.FOR
 
+    @classmethod
+    def from_params(cls, for_separator, values, variables, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
```

Comment:
Not sure about the order of arguments. `variables, for_separator, values` would be the same order as in the data, but using `variables, values, for_separator='IN'` would allow using a default value with the separator.

Reply:
`variables, values, for_separator='IN'` sounds handy so I will use it.

Reply:
Just realized the name of the property accessing this value is `flavor` and that might be a good name for the argument as well. The token type is `FOR_SEPARATOR` so the current `for_separator` is fine as well.

Reply:
When debugging the first thing people see is most likely `flavor` rather that type of the token (`FOR_SEPARATOR`). And if they check source code, they will see variables, values and flavor properties first. So I will change argument name to follow it 👍